src/reader/mod.rs

```diff
@@ -192,12 +192,8 @@ impl<T: ZarrIterator> ZarrRecordBatchReader<T> {
         let mut arrs: Vec<ArrayRef> = Vec::with_capacity(self.meta.get_num_columns());
         let mut fields: Vec<FieldRef> = Vec::with_capacity(self.meta.get_num_columns());
 
-        // the sort below is important, because within a zarr store, the different arrays are
```

Comment:
Turns out this is a proper bug, because I already thought about this before (and then I forgot about it), but the way I tried to enforce ordering wasn't quite right.